src/lwc/geListView/geListView.js

```diff
@@ -0,0 +1,714 @@
+                columnHeader[COLUMN_NAME_INFO.fieldApiName] = generateId();
+                columnHeader[COLUMN_LIST_NAME_INFO.fieldApiName] = this.listName;
+            }
+
+            columnHeader[COLUMN_FIELD_API_NAME_INFO.fieldApiName] = fieldApiName;
+            columnHeader[COLUMN_INDEX_INFO.fieldApiName] = index;
+
+            return columnHeader;
+        });
+    }
+
+    /*******************************************************************************
+    * @description Method handles setting up the lightning-datatable columns based
+    * on the displayColumn that's provided by the getListUi api.
+    *
+    * @param {object} displayColumns: Data describing the column headers for a list
+    * view
+    */
+    setDatatableColumns(displayColumns) {
+        this.columnEntriesByName = {};
+        this.columns = [];
+
+        displayColumns.forEach(column => {
+            let fieldApiName = column.fieldApiName;
+
+            // Special case for relationship references e.g. 'CreatedBy.Name'
+            // so we can display the Name prop of the reference.
+            if (column.fieldApiName.includes(`.${NAME}`)) {
+                fieldApiName = column.fieldApiName.split('.')[0];
+            }
+
+            // Gotta create a new column object with correct properties
+            // because for some reason lightning-datatable
+            // requires the property 'fieldName', but we're given
+            // the property 'fieldApiName' by the getListUi api
+            let columnEntry = {
+                label: column.label,
+                fieldName: fieldApiName,
+                fieldApiName: column.fieldApiName,
+                sortable: column.sortable
+            }
+
+            // Turn fields in the 'Name' column into URLs
+            if (fieldApiName === NAME) {
+                columnEntry.type = URL;
+                columnEntry.fieldName = URL;
+                columnEntry.typeAttributes = {
+                    label: {
+                        fieldName: fieldApiName
+                    },
+                    target: _SELF
+                }
+            }
+
+            this.columns = [...this.columns, columnEntry];
+            this.columnEntriesByName[columnEntry.fieldName] = columnEntry;
+        });
+    }
+
+    /*******************************************************************************
+    * @description Method sets the actions for the lightning-datatable based on
+    * the public property 'actions'.
+    */
+    setDatatableActions() {
+        if (this.actions) {
+            this.columns = [...this.columns, {
+                type: 'action',
+                typeAttributes: { rowActions: this.actions },
+            }];
+        }
+    }
+
+    /*******************************************************************************
+    * @description Method builds a SOQL query based on the currently selected list
+    * view's describe info.
+    */
+    buildSoqlQuery(selectedFieldApiNames) {
```

Comment:
I have a few questions here:
1) first of all, should we building the SOQL query in JS as opposed to sending the needed info into the call to the Apex method and building the query using the tools available there already?
2) Is there a potential security risk for building the SOQL query in JS? 
3) If we DO want to build the SOQL in javascript, should this be moved to a more commonly accessible utility not just built into the list view component?

Reply:
Yeah, this was also the most if-y section of this work in my mind. This bit was left over from when I was relying on the metadata from the uiListApi and decided to make the most minimal amount of change and still have it work.

The only input into this query builder that's end-user accessible is for the query's select clause via their selection of column headers in that dual-listbox modal. I _want_ to say it's relatively safe as we're not pulling strings/values that's directly written by the user.

Regardless I totally get the concern and am down to move this logic into the apex method if that's what we want 👍

Reply:
@kenlewis-sfdo This was what I wanted to get a second opinion on - This is replicating some SOQL query-building function that already exists in apex within the Javascript controller, and I'm wondering both if this is a good idea, and if we would want to bundle this as its own service component if we did like it within the front end. Or any other thoughts or concerns!

Reply:
If we move ahead with keeping it, a service component would be a good idea.  But my opinion would be to lean away from replicating the SOQL query-building functionality in JS.  Since we already have it in Apex and this component already imports it (`FORM_ServiceGiftEntry.retrieveRecords`), I wouldn't keep it unless we verify that the performance gain outweighs the added maintenance having query-building logic in multiple places.  I would pass the fields as an array to the Apex method and maybe add a `withOffset` method to the `UTIL_Query` class to implement infinite scroll on the `lightning-datatable`.

Reply:
I think that's how I'm leaning as well (or at least how I'm leaning now that I know that's how Ken is leaning!) ... but it feels better to only build that code once and use it in Apex where it's already developed.